test/topology_raft_disabled/test_raft_upgrade.py

```diff
@@ -215,6 +216,19 @@ async def test_raft_upgrade_with_node_remove(manager: ManagerClient, random_tabl
     hosts = await wait_for_cql_and_get_hosts(cql, others, time.time() + 60)
     logging.info(f"Driver reconnected, hosts: {hosts}")
 
+    # Wait until the alive nodes mark each other as alive - otherwise remove_node may fail.
+    async def only_srv1_dead() -> Optional[bool]:
+        expected = [srv1.ip_addr]
+        for srv in others:
+            for shard in range(NUM_SHARDS):
+                down_endpoints = await manager.api.get_down_endpoints(srv.ip_addr, shard = shard)
```

Comment:
Why do you need to specify the shard anyway? Can't you use shard 0 in all cases?

Reply:
No.

[Waiting on shard 0 only does not guarantee
that other shards have the same information yet (it takes a while for
this information to replicate between shards).
](https://github.com/scylladb/scylladb/pull/12173#issuecomment-1351805399)